Backend/app/Http/Controllers/UserController.php

```diff
@@ -16,6 +21,152 @@ public function getCurrentUser()
             return response()->json(["user" => $user], 200);
         }
 
-        return response()->json(["message" => "Użytkownik niezalogowany"], 401);
+        return response()->json(["message" => "User not Auth"], 401);
+    }
+
+    public function getAllUsers()
+    {
+        $users = User::all();
+
+        return response()->json($users);
+    }
+
+    public function addNewUser(UserRequest $request)
+    {
+        $user = User::create([
+            "name" => $request->name,
+            "email" => $request->email,
+            "password" => Hash::make($request->password),
+            "role" => $request->role,
+        ]);
+
+        return response()->json($user, 201);
+    }
+
+    public function deleteUser($id)
+    {
+        $user = User::find($id);
+
+        if (!$user) {
+            return response()->json(["message" => "User not found"], 404);
+        }
+
+        $user->delete();
+
+        return response()->json(["message" => "User deleted"]);
+    }
+
+    public function changeName(ChangeNameRequest $request, $id)
+    {
+        $user = User::find($id);
+
+        if ($user->name !== $request->oldName) {
+            return response()->json(["message" => "Old name not match."], 400);
+        }
+
+        $user->update([
+            "name" => $request->newName,
+        ]);
+
+        return response()->json($user);
+    }
```

Comment:
Is it necessary for the user to enter their current name to change it? I don't think so

Reply:
When i first looked at this fragment of code, i didn't understand why he did it that way. 

I guess it is more of the safe check for admin so that he get console message to make sure he edit right person and not some other user.